public/logic.js

```diff
@@ -1,155 +1,130 @@
-      window.location.pathname = "/consent"
+    var buttonsArr = Array.from(document.querySelectorAll("button[id]"))
+    buttonsArr.forEach(function (item) {
+      item.addEventListener('click', function () {
+        sessionStorage.setItem(path, item.innerHTML)
+        nextQuestion()
+      })
     })
   }
 
-  // obtain consent
-  if (path === 'consent') {
-    var survey = sessionStorage.getItem('select-survey')
-    var yesConsent = document.getElementById('yes-consent')
-    yesConsent.addEventListener('click', function () {
-      sessionStorage.setItem(path, 'yes')
-      if (survey==='pre-session') {
-        window.location.pathname = "/today"
-      } else {
-        window.location.pathname = "/understand"
-      }
-    })
-
-    var noConsent = document.getElementById('no-consent')
-    noConsent.addEventListener('click', function () {
-      sessionStorage.setItem(path, 'no')
-      window.location.pathname = '/thank-you'
+  if (path === 'confirm-survey' || path === 'thank-you') {
+    var myButton = Array.from(document.querySelectorAll('button'))
+    myButton[0].addEventListener('click', function () {
+      nextQuestion()
     })
   }
 
-  if (path==='thank-you'){
-    var okSubmit = document.getElementById('submit-survey-button')
-    okSubmit.addEventListener('click', function(){
-      window.location.pathname = '/login'
+  if(path === 'login') {
+    var myButton = document.getElementById('submit-login-button')
+    //E-mail validation in the future refactoring
+    myButton.addEventListener('click', function () {
+    sessionStorage.setItem('e-mail', JSON.stringify(document.getElementById('clinician-email-input').value))
+    nextQuestion()
     })
   }
+}
 
-  if (path==='login'){
-    var submitLoginDetails = document.getElementById('submit-login-button')
-    submitLoginDetails.addEventListener('click', function(){
-      var clinicianEmail = document.getElementById('clinician-email-input').value
-      sessionStorage.setItem('clinician-email', clinicianEmail)
-      window.location.pathname = '/results'
-    })
-  }
+buttonListeners();
 
-  // pre-session responses
-  if (path === 'today' ||
-    path === 'last-week' ||
-    path === 'home' ||
-    path === 'friends' ||
-    path === 'family' ||
-    path === 'school' ||
-    path === 'play' ||
-    path === 'next-week') {
-    var veryGood = document.getElementById('very-good-option')
-    veryGood.addEventListener('click', function () {
-      sessionStorage.setItem(path, 'very good')
-      nextQuestion()
-    })
+function answerListeners() {
+  var pathsArr = ['understand', 'like', 'help', 'come-again', 'today', 'last-week', 'friends', 'family', 'school', 'play', 'next-week']
+  var answersArr = Array.from(document.querySelectorAll("figure[id]"))
 
-    var good = document.getElementById('good-option')
-    good.addEventListener('click', function () {
-      sessionStorage.setItem(path, 'good')
-      nextQuestion()
+  if (pathsArr.includes(path.toString())) {
```

Comment:
array.includes is es6. Nasty workarout:
```js
array.indexOf(elem) !== 0
```

Reply:
@des-des Thank you for the idea. Though I think you accidentally typed '0' instead of '-1'. In order to work it has to be `array.indexOf(elem) !== -1`